base/linalg/bidiag.jl

```diff
@@ -12,7 +12,7 @@ type Bidiagonal{T} <: AbstractMatrix{T}
         new(dv, ev, isupper)
     end
 end
-Bidiagonal{T}(dv::AbstractVector{T}, ev::AbstractVector{T}, isupper::Bool) = Bidiagonal{T}(dv, ev, isupper)
+Bidiagonal{T}(dv::AbstractVector{T}, ev::AbstractVector{T}, isupper::Bool) = Bidiagonal{T}(collect(dv), collect(ev), isupper)
```

Comment:
I think a better fix for this is simply to remove the `{T}` part of the inner constructor for `Bidiagonal`. Then the conversion is done automatically, which is a bit clearner.